src/System.Linq/src/System/Linq/Enumerable.cs

```diff
@@ -2865,6 +2978,23 @@ public bool Contains(TKey key)
                 } while (g != _lastGrouping);
             }
         }
+        
+        public IGrouping<TKey, TElement>[] ToArray()
```

Comment:
I missed this in the initial review... `Lookup<TKey, TElement` is a public class, so this is adding a new public `ToArray` method. Instead, this new method should be implemented explicitly (private).